aws-transit-gateway/scripts/generate_tf_var_input.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+cluster_id=$1
+
+if [ $# -ne 1 ]; then
+   echo "Error: Usage: $0 <redpanda-id>"
+   exit 1
+fi
+
+if [ "$CLOUD_CLIENT_ID" = "" -o "$CLOUD_CLIENT_SECRET" = "" ]; then
+   echo "Error: Missing the env variables CLOUD_CLIENT_ID and CLOUD_CLIENT_SECRET"
+   exit 2
+fi
+
+PUBLIC_API_ENDPOINT=${PUBLIC_API_ENDPOINT:-"https://api.redpanda.com"}
+PUBLIC_API_AUTH_ENDPOINT=${PUBLIC_API_AUTH_ENDPOINT:-"https://auth.prd.cloud.redpanda.com"}
+PUBLIC_API_AUTH_AUDIENCE=${PUBLIC_API_AUTH_AUDIENCE:-"cloudv2-production.redpanda.cloud"}
+
+token=`curl -s --request POST \
+         --url $PUBLIC_API_AUTH_ENDPOINT/oauth/token \
+         --header 'content-type: application/x-www-form-urlencoded' \
+         --data grant_type=client_credentials \
+         --data client_id=$CLOUD_CLIENT_ID \
+         --data client_secret=$CLOUD_CLIENT_SECRET \
+         --data audience=$PUBLIC_API_AUTH_AUDIENCE | jq .access_token | sed 's/"//g'`
```

Comment:
```suggestion
if [ "$REDPANDA_CLIENT_ID" = "" -o "$REDPANDA_CLIENT_SECRET" = "" ]; then
   echo "Error: Missing the env variables REDPANDA_CLIENT_ID and REDPANDA_CLIENT_SECRET"
   exit 2
fi

PUBLIC_API_ENDPOINT=${PUBLIC_API_ENDPOINT:-"https://api.redpanda.com"}
PUBLIC_API_AUTH_ENDPOINT=${PUBLIC_API_AUTH_ENDPOINT:-"https://auth.prd.cloud.redpanda.com"}
PUBLIC_API_AUTH_AUDIENCE=${PUBLIC_API_AUTH_AUDIENCE:-"cloudv2-production.redpanda.cloud"}

token=`curl -s --request POST \
         --url $PUBLIC_API_AUTH_ENDPOINT/oauth/token \
         --header 'content-type: application/x-www-form-urlencoded' \
         --data grant_type=client_credentials \
         --data client_id=$REDPANDA_CLIENT_ID \
         --data client_secret=$REDPANDA_CLIENT_SECRET \
         --data audience=$PUBLIC_API_AUTH_AUDIENCE | jq .access_token | sed 's/"//g'`
```

Perhaps we can use this instead per https://docs.redpanda.com/redpanda-cloud/get-started/cluster-types/byoc/aws/vpc-byo-aws/#authenticate-with-redpanda-cloud

Reply:
Ok